app/bundles/PointBundle/Tests/Unit/Model/PointModelTest.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\PointBundle\Tests\Unit\Model;
+
+use Doctrine\ORM\EntityManager;
+/**
+ * @deprecated to be removed in 3.0
+ */
+use Mautic\CoreBundle\Factory\MauticFactory;
+use Mautic\CoreBundle\Helper\CoreParametersHelper;
+use Mautic\CoreBundle\Helper\IpLookupHelper;
+use Mautic\CoreBundle\Helper\UserHelper;
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
+use Mautic\CoreBundle\Translation\Translator;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\LeadBundle\Model\LeadModel;
+use Mautic\LeadBundle\Tracker\ContactTracker;
+use Mautic\PointBundle\Entity\Point;
+use Mautic\PointBundle\Entity\PointRepository;
+use Mautic\PointBundle\Model\PointGroupModel;
+use Mautic\PointBundle\Model\PointModel;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+class PointModelTest extends TestCase
+{
+    private PointModel $pointModel;
+    /**
+     * @var PointRepository&\PHPUnit\Framework\MockObject\MockObject
+     */
+    private PointRepository $pointRepository;
+    /**
+     * @var IpLookupHelper&\PHPUnit\Framework\MockObject\MockObject
+     */
+    private IpLookupHelper $ipLookupHelper;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        // Create mock for the repository
+        $this->pointRepository = $this->createMock(PointRepository::class);
+
+        // Create mock for IP lookup helper
+        $this->ipLookupHelper = $this->createMock(IpLookupHelper::class);
+
+        // Create all other required mocks
+        $requestStack         = $this->createMock(RequestStack::class);
+        $leadModel            = $this->createMock(LeadModel::class);
+        $mauticFactory        = $this->createMock(MauticFactory::class);
```

Comment:
This will need a PHPSTAN ignore line comment. Or if it is possible, let's rebase on the 6.x branch which doesn't have the MauticFactory anymore. This will cause conflict once we'll release the M5 and try to merge it to M6 and 7. But if you need this in M5 then let's leave it be and deal with the conflicts.